pageserver/src/tenant/timeline/compaction.rs

```diff
@@ -1115,7 +1115,13 @@ impl Timeline {
         // Under normal circumstances, we will accumulate up to compaction_upper_limit L0s of size
         // checkpoint_distance each.  To avoid edge cases using extra system resources, bound our
         // work in this function to only operate on this much delta data at once.
-        let delta_size_limit = self.get_compaction_upper_limit() as u64
+        //
+        // In general, compaction_threshold should be <= compaction_upper_limit, but in case that
+        // the constraint is not respected, we use the larger of the two.
+        let delta_size_limit = std::cmp::max(
+            self.get_compaction_upper_limit(),
```

Comment:
nit: minor preference for moving this into `get_compaction_upper_limit()`.

Reply:
the existing pattern for all these `get_$tenant_config_item` functions is to do nothing but merge the global default and the per-tenant overrides.

would prefer we stick to that pattern (and DRY that at some point)

Reply:
actually, we now have some exceptions like `get_l0_flush_delay_threshold` :) I'm neutral on this so I'd keep it as-is now.

Reply:
I'm fine to ship this as-is.